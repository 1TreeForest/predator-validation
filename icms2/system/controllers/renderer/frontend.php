<?php
class renderer extends cmsFrontend {

    private $middleware;

    public function __construct($request){

        parent::__construct($request);

        $this->middleware = $this->request->get('middleware');
    }

    public function get() {

        $event_name = 'middleware_'.$this->middleware;

        $class_name = 'on' . string_to_camel('_', $this->name) . string_to_camel('_', $event_name);

        if (!class_exists($class_name, false)) {

            $hook_file = $this->root_path . 'hooks/' . $event_name . '.php';

            if (!is_readable($hook_file)) {
                cmsCore::error(ERR_FILE_NOT_FOUND . ': ' . str_replace(PATH, '', $hook_file));
            }

            include_once $hook_file;
        }

        return new $class_name($this);
    }

    public function render() {

        $params = func_get_args();

        return call_user_func_array([$this->get(), 'run'], $params);
    }

}
