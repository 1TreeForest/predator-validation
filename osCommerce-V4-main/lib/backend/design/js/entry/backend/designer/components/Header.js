import React from 'react';
import { connect } from 'react-redux'
import Nav from 'react-bootstrap/Nav';
import NavDropdown from 'react-bootstrap/NavDropdown';
import globals from 'src/globals';
import { toggleSidebar } from '../reducers/layout.actions';

class Header extends React.Component {

    render() {
        return <div className="header">
            <a className="logo" href={globals('baseUrl')}>
                <img src={globals(['wl', 'companLogoUrl'])} alt="logoAlt" />
            </a>

            <div className={'toggle-sidebar ' + (this.props.toggleSidebar ? 'active' : '')} onClick={this.props.onToggleSidebar}></div>

            <Nav>
                <Nav.Item>
                    <Nav.Link href={globals('frontendUrl')}>{globals(['tr', 'TEXT_VIEW_SHOP'])}</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                    <Nav.Link href={globals(['wl', 'servicesUrl'])}>{globals(['wl', 'servicesText'])}</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                    <Nav.Link href={globals(['wl', 'supportUrl'])}>{globals(['wl', 'supportText'])}</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                    <Nav.Link href={globals(['wl', 'contactUsUrl'])}>{globals(['wl', 'contactUsText'])}</Nav.Link>
                </Nav.Item>

                <NavDropdown title={globals(['tr', 'TEXT_MY_ACCOUNT'], 'My Account')} id="nav-dropdown">
                    <NavDropdown.Item eventKey="4.1">
                        <div className="admin-view">
                            {globals(['adminData', 'avatar']) ?
                                <img src={globals('DIR_WS_CATALOG_IMAGES') + globals(['adminData', 'avatar'])}/>
                                :
                                <i className="icon-user"></i>
                            }
                            <span className="full-name">{globals(['adminData', 'firstname'])}  {globals(['adminData', 'lastname'])}</span>
                        </div>
                    </NavDropdown.Item>
                    <NavDropdown.Item href={globals('baseUrl') + 'adminaccount'}>{globals(['tr', 'TEXT_MY_ACCOUNT'])}</NavDropdown.Item>
                    <NavDropdown.Item href={globals('baseUrl') + 'logout'}>{globals(['tr', 'TEXT_HEADER_LOGOUT'])}</NavDropdown.Item>
                </NavDropdown>
            </Nav>

        </div>;
    }
}

const mapStateToProps = (state, ownProps) => ({
    toggleSidebar: state.layout.toggleSidebar
})

const mapDispatchToProps = (dispatch, ownProps) => ({
    onToggleSidebar: () => dispatch(toggleSidebar())
})

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Header)