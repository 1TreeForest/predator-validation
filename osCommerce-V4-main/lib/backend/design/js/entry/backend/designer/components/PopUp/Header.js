import React from 'react';

export default class Header extends React.Component {
    constructor(props) {
        super(props);

    }

    render() {
        return (
            <div className="popup-heading">
                {this.props.icons && Array.isArray(this.props.icons) && this.props.icons.length > 0 ? (
                    <div className="icons">
                        {this.props.icons.map(item => item)}
                    </div>
                ) : ''}
                {this.props.children}
            </div>
        )
    }
};