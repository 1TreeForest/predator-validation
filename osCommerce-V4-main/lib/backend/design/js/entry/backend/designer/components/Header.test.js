import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { act } from "react-dom/test-utils";

import Header from './Header';


const mockStore = configureStore([]);

describe('Header Connected', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            layout: {
                sidebarWidth: 265,
                toggleSidebar: true,
                menuType: 'advanced',
                pageTitle: 'Themes'
            },
            topButtons: {
                addTheme: true
            },
            toolWindows: {
                addTheme: false
            },
            designer: {
            }
        });

        component = renderer.create(
            <Provider store={store}>
                <Header />
            </Provider>
        );
    });

    it('Header snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on button click', () => {

        const instance = component.getInstance()

        console.log(component.root);
        /*renderer.act(() => {
            component.root.findAllByProps('onToggleSidebar').props.onClick();
        });

        renderer.act(() => {
            component.root.findByType('input')
                .props.onChange({ target: { value: 'some other text' } });
        });

        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            //myAction({ payload: 'some other text' })
        );*/
    });
});
