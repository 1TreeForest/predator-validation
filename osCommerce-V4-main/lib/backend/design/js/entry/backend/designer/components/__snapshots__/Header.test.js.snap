// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="header"
>
  <a
    className="logo"
    href={null}
  >
    <img
      alt="logoAlt"
      src={null}
    />
  </a>
  <div
    className="toggle-sidebar active"
    onClick={[Function]}
  />
  <div
    className="nav"
    onKeyDown={[Function]}
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      />
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      />
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      />
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      />
    </div>
    <div
      className="dropdown nav-item"
      onKeyDown={[Function]}
    >
      <a
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle nav-link"
        data-rb-event-key={null}
        href="#"
        id="nav-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        My Account
      </a>
    </div>
  </div>
</div>
`;
